spec/support/medium_controller_spec.rb

```diff
@@ -0,0 +1,79 @@
+require 'rails_helper'
+require 'pry'
+
+RSpec.shared_examples "media" do
+  describe "controller validations" do
+    describe "GET 'index'" do
+      it "is successful" do
+        get :index
+        expect(response).to have_http_status 200
+      end
+    end
+
+    describe "GET 'new'" do
+      it "renders new view" do
+        get :new
+        expect(subject).to render_template :new
+      end
+    end
+
+    describe "GET 'edit'" do
+      it "renders edit view" do
+        get :edit, id: item.id
+        expect(subject).to render_template :edit
+      end
+    end
+
+
+    describe "GET 'show'" do
+      it "renders the show view" do
+        get :show, id: item.id
+        expect(subject).to render_template :show
+      end
+    end
+
+    describe "PATCH 'upvote'" do
+     it "increments ranking" do
+       patch :upvote, id: item.id
+       item.reload
+       expect(item.ranking).to eq 1
```

Comment:
In order to make this more general, we could test only that the `ranking` value was incremented by 1, rather than was set directly to 1:
```ruby
it "increments ranking" do
  original_ranking = item.ranking

  patch :upvote, id: item.id

  item.reload
  expect(item.ranking).to eq original_ranking + 1
end
```